### ISSUE TITLE
--ignore dir doesn't work well for changes made by command

### ISSUE THREAD

**Issue by c02y** [watchexec/watchexec#284]
Please delete this template text before filing, but you _need_ to include the following:

- Watchexec's version: `1.19.0` installed from Arch Linux repo
- The OS you're using: Arch Linux
- A log with `-vvv` (if it has sensitive info you can email it at felix@passcod.name — do that _after_ filing so you can reference the issue ID)
 ```This output becomes thousands of lines in a second, and the command runs repeatedly, the output is full of color code if I output the result into a file, so it is unreadable, unable to paste here.```
- A sample command that you've run that has the issue

Create a simple project structure like this
```
- CMakeLists.txt
- test.cpp
```

Contents like this:
```
$ cat CMakeLists.txt
cmake_minimum_required(VERSION 3.18)
project(test)

add_executable(test test.cpp)

$ cat test.cpp
#include <iostream>

int main(int argc, char *argv[])
{
        std::cout << "hello word" << std::endl;
        return 0;
}
```
Run command like this: 
```
watchexec -i build "cmake -B build"
```

it will keep running repeatedly forever, since files in build are changed every time you run the command.

But (suggested by https://github.com/watchexec/watchexec/issues/254#issuecomment-1028413927)
```
watchexec -i "build/**" "cmake -B build"
```
works very well. 

Thank you

FYI: I think this is related to https://github.com/watchexec/watchexec/issues/40 and https://github.com/watchexec/watchexec/issues/123, and `--watch-when-idle` doesn't work at all (if this is the flag for this purpose)

**Comment by passcod** [comment]
I can strip the color codes myself if you send me a file, it's fine. It's likely the same issue as #254, and having another log will help find a good solution.

**Comment by c02y** [comment]
Output of:
```
watchexec -i build -vvv "cmake -B build"
```
[log.txt](https://github.com/watchexec/watchexec/files/8782699/log.txt)

I only run it and `Ctrl-c` in less than a second, since it contains over 10k of lines after a second, if this log is not enough, please let me know.

And you can follow the easy steps I provided to reproduce this log.

**Comment by passcod** [comment]
Thanks, I'll have a look tonight!

**Comment by passcod** [comment]
Alright, this is indeed a duplicate of #254, and will likely be solved with the work for #250.

At its core, this is an issue with how the underlying `globset` library processes and interprets these patterns (it resolves it to `**/build`); the work for #250 is essentially a rewrite of the lower levels of ignore handling, due to other similar subtle issues with the `globset` and `ignore` crates.

Obviously that's a decent chunk of work; in the meantime using `**/build/**` or `build/**` can be used as workaround.
